src/pages/AdvancedSearch/ResultsBlock/ResultsBlock.tsx

```diff
@@ -0,0 +1,99 @@
+import {
+  CodeSnippet,
+  CodeSnippetBlockAppearance,
+  Spinner,
+} from "@canonical/react-components";
+import { useEffect, useState } from "react";
+
+import { actions as jujuActions } from "store/juju";
+import {
+  getCrossModelQueryLoaded,
+  getCrossModelQueryLoading,
+  getCrossModelQueryResults,
+} from "store/juju/selectors";
+import { useAppDispatch, useAppSelector } from "store/store";
+
+export enum TestId {
+  LOADING = "loading",
+}
+
+enum CodeSnippetView {
+  TREE = "tree",
+  JSON = "json",
+}
+
+const ResultsBlock = (): JSX.Element | null => {
+  const dispatch = useAppDispatch();
+  const isCrossModelQueryLoading = useAppSelector(getCrossModelQueryLoading);
+  const isCrossModelQueryLoaded = useAppSelector(getCrossModelQueryLoaded);
+  const crossModelQueryResults = useAppSelector(getCrossModelQueryResults);
+  const [codeSnippetView, setCodeSnippetView] = useState<CodeSnippetView>(
+    CodeSnippetView.TREE
+  );
+  const codeSnippetContent = {
+    tree: "",
+    json: JSON.stringify(crossModelQueryResults, null, 2),
+  };
+
+  useEffect(
+    () => () => {
+      // Clear cross-model query results when component is unmounted.
+      dispatch(jujuActions.clearCrossModelQuery());
+    },
+    [dispatch]
+  );
+
+  if (!isCrossModelQueryLoaded) {
+    return null;
+  }
+
+  if (isCrossModelQueryLoading) {
+    return (
+      <div className="u-align--center">
+        <div className="u-sv3">
+          <hr />
+        </div>
+        <Spinner data-testid={TestId.LOADING} />
+      </div>
+    );
+  }
+
+  return (
+    <>
+      <hr />
```

Comment:
I think this line is only necessary when displaying the spinner so that it's not floating in space.

Reply:
I'll fix this up in my next branch.